os/fs/driver/mtd/smart.c

```diff
@@ -4613,9 +4614,13 @@ static int smart_allocsector(FAR struct smart_struct_s *dev, unsigned long reque
 	 * ensure we keep enough reserved free sectors to perform garbage
 	 * collection as it involves moving sectors from blocks with
 	 * released sectors into blocks with free sectors, then
-	 * erasing the vacated block. */
+	 * erasing the vacated block.
+	 * Check if the number of free sectors is less tha a certain threshold (4%)
+	 */
+	if (dev->freesectors < (dev->availSectPerBlk * dev->neraseblocks) / 25) {
```

Comment:
Regarding the computational overhead of floating division, how about 6.3% threshold, i.e., dev->availSectPerBlk * dev->neraseblocks) >> 4?

Reply:
I agree, using this will be advantageous
And 6.3% works as our limit well enough

Reply:
Updated this PR
Sorry for the delay.